File: ContactService/ContactService.cpp
// ContactService.cpp

#include "ContactService.hpp"


string ContactService::addContact(const string& firstName,
	const string& lastName,
	const string& number,
	const string& address) {
	Contact c(firstName, lastName, number, address);
	contacts_.emplace(c.getContactID(), c); // Insert contact into the unordered map with contactID as the key
	return c.getContactID();
}

void ContactService::deleteContact(const string& contactID) {
	if (contacts_.erase(contactID) == 0) { // erase returns the number of elements removed, which will be 0 if the contactID was not found
		throw NotFoundError("Contact with ID " + contactID + " not found.");
		}
	}


Contact& ContactService::findOrThrow(const string& contactID) {
	auto it = contacts_.find(contactID);
	if (it == contacts_.end()) {
		throw NotFoundError("Contact with ID " + contactID + " not found.");
	}
	return it->second;
}

const Contact& ContactService::findOrThrow(const string& contactID) const {
	auto it = contacts_.find(contactID);
	if (it == contacts_.end()) {
		throw NotFoundError("Contact with ID " + contactID + " not found.");
	}
	return it->second;
}

void ContactService::updateFirstName(const string& contactID, const string& updated) {
	Contact& contact = findOrThrow(contactID);
	contact.setFirstName(updated);
}

void ContactService::updateLastName(const string& contactID, const string& updated) {
	Contact& contact = findOrThrow(contactID);
	contact.setLastName(updated);
}

void ContactService::updatePhoneNumber(const string& contactID, const string& updated) {
	Contact& contact = findOrThrow(contactID);
	contact.setPhoneNumber(updated);
}

void ContactService::updateAddress(const string& contactID, const string& updated) {
	Contact& contact = findOrThrow(contactID);
	contact.setAddress(updated);
}

const Contact& ContactService::getContactorThrow(const string& contactID) const {
	return findOrThrow(contactID);
}

size_t ContactService::size() const {
	return contacts_.size();
}

File: Artifact Service/ContactService.java


package contact; // Asserts that the class is in the contact package. 

import java.util.ArrayList; // imports array functionality from java utility. 





public class ContactService {
	public ArrayList<Contact> contactList = new ArrayList<Contact>(); // Creates a new array which will store all contact information. 
	
	// Inserts each contact "instance" into the list until no instances are left. 
	public void displayContactList() {
		for (int i = 0; i < contactList.size(); i++) {
			System.out.println("\t Contact ID: " + contactList.get(i).getContactID());
			System.out.println("\t First Name: " + contactList.get(i).getFirstName());
			System.out.println("\t Last Name: " + contactList.get(i).getLastName());
			System.out.println("\t Phone Number: " + contactList.get(i).getNumber());
			System.out.println("\t Address: " + contactList.get(i).getAddress()+ "\n");
		}
	}
	// Creates a function to add contacts. 
	public void addContact(String firstName, String lastName, String number, String address) {
		Contact contact = new Contact(firstName, lastName, number, address);
		contactList.add(contact);
		
	}
	// Creates a function to delete contacts. 
	public void deleteContact(String contactID) {
		for (int i = 0; i < contactList.size(); i++) {
			if (contactList.get(i).getContactID().equals(contactID)) {
				contactList.remove(i);
				break;
			}
			if (i == contactList.size() - 1) {
				System.out.println("Contact ID: " + contactID + "not found."); // Reports if the complier cannot find the contactID searched for by the user. 
			}
		}
	}
	// Updates the first name of a contact. 
	public void updateFirstName(String updatedString, String contactID) {
		for (int i = 0; i < contactList.size(); i++) {
			if (contactList.get(i).getContactID().equals(contactID)) {
				contactList.get(i).setFirstName(updatedString);
				break;
			}
			if (i == contactList.size() - 1) {
				System.out.println("Contact ID: " + contactID + "not found.");
			}
		}
	}
	// Updates the last name of a contact. 
	public void updateLastName(String updatedString, String contactID) {
		for (int i = 0; i < contactList.size(); i++) { 
			if (contactList.get(i).getContactID().equals(contactID)) {
				contactList.get(i).setLastName(updatedString);
				break;
			}
			if (i == contactList.size() - 1) {
				System.out.println("Contact ID: " + contactID + " not found.");
			}
		}
	}
	// Updates the number of a contact. 
	public void updateNumber(String updatedString, String contactID) {
		for (int i = 0; i < contactList.size(); i++) {
			if (contactList.get(i).getContactID().equals(contactID)) {
				contactList.get(i).setNumber(updatedString);
				break;
			}
			if (i == contactList.size() - 1) {
				System.out.println("Contact ID: " + contactID + " not found.");
			}
		}
	}
	// Updates the address of a contact. 
	public void updateAddress(String updatedString, String contactID) {
		for (int i = 0; i < contactList.size(); i++) {
			if (contactList.get(i).getContactID().equals(contactID)) {
				contactList.get(i).setAddress(updatedString);
				break;
			}
			if (i == contactList.size() - 1) {
				System.out.println("Contact ID: " + contactID + " not found.");
			}
		}
	}
}
